src/snowcli/cli/nativeapp/utils.py

```diff
@@ -1,15 +1,58 @@
-from typing import Callable, Optional
+from dataclasses import dataclass
+from sys import stdin, stdout
+from typing import Callable, List, Optional
 
+import typer
 from snowflake.connector.cursor import DictCursor
 
 
-def find_row(cursor: DictCursor, predicate: Callable[[dict], bool]) -> Optional[dict]:
+def _rows_generator(cursor: DictCursor, predicate: Callable[[dict], bool]):
+    return (row for row in cursor.fetchall() if predicate(row))
+
+
+def find_all_rows(cursor: DictCursor, predicate: Callable[[dict], bool]) -> List[dict]:
+    return list(_rows_generator(cursor, predicate))
+
+
+def find_first_row(
+    cursor: DictCursor, predicate: Callable[[dict], bool]
+) -> Optional[dict]:
     """Returns the first row that matches the predicate, or None."""
-    return next(
-        (row for row in cursor.fetchall() if predicate(row)),
-        None,
-    )
+    return next(_rows_generator(cursor, predicate), None)
 
 
 def needs_confirmation(needs_confirm: bool, auto_yes: bool) -> bool:
     return needs_confirm and not auto_yes
+
+
+class UserConfirmationPolicy:
+    def __init__(self, is_force, is_interactive_mode, confirm_with_user):
+        self.is_force = is_force
+        self.is_interactive_mode = is_interactive_mode and not self.is_force
+        self.confirm_with_user = confirm_with_user
+
+    def requires_user_confirmation(self):
+        return not self.is_force
+
+
+@dataclass
+class Prompts:
+    give_info_and_ask: str
+    do_not_continue_message: str
+    error_message: str
+
+
+def ask_user_confirmation(
+    user_confirmation_policy: UserConfirmationPolicy, prompts: Prompts
+):
+    if user_confirmation_policy.requires_user_confirmation():
```

Comment:
Might be useful to see it as a policy that controls the execution of a task, not a policy that controls asking the user. Asking is a mechanism through which we allow a task to execute. The ultimate goal is task execution. If you see it this way, it becomes possible to have a policy that allows certain tasks to execute while confirming others (e.g. ok to overwrite files, but not to drop a database)